packages/samples/headless-react/src/components/standalone-search-box/standalone-search-box.class.tsx

```diff
@@ -0,0 +1,89 @@
+import {Component, ContextType} from 'react';
+import {
+  buildStandaloneSearchBox,
+  StandaloneSearchBox as HeadlessStandaloneSearchBox,
+  StandaloneSearchBoxState,
+  Unsubscribe,
+} from '@coveo/headless';
+import {AppContext} from '../../context/engine';
+
+export class StandaloneSearchBox extends Component<
+  {},
+  StandaloneSearchBoxState
+> {
+  static contextType = AppContext;
+  context!: ContextType<typeof AppContext>;
+
+  private controller!: HeadlessStandaloneSearchBox;
+  private unsubscribe: Unsubscribe = () => {};
+
+  componentDidMount() {
+    this.controller = buildStandaloneSearchBox(this.context.engine!, {
+      options: {redirectionUrl: 'https://mywebsite.com/search'},
+    });
+    this.updateState();
+
+    this.unsubscribe = this.controller.subscribe(() => this.updateState());
+  }
+
+  componentWillUnmount() {
+    this.unsubscribe();
+  }
+
+  componentDidUpdate() {
+    if (!this.state?.redirectTo) {
+      return;
+    }
+    window.location.href = this.state.redirectTo + this.getURIParameters();
+  }
+
+  private updateState() {
+    this.setState(this.controller.state);
+  }
+
+  private getURIParameters() {
+    const {redirectTo, value} = this.state;
+    const query = encodeURIComponent(value);
+    // The query pipeline can trigger different redirection URLs, as documented here: https://docs.coveo.com/en/1458/tune-relevance/trigger-query-pipeline-feature
+    // Unless a redirect trigger is added, you won't need such conditions. These redirections are only verified here as an example.
```

Comment:
I'm not sure what to write here... Opinions?

@fbeaudoincoveo @kaitlynkatz @jpmarceau

Reply:
I would vote to remove that explanation for the sample that are meant to appear in the controller documentation.

Those more complex "how to actually do redirection properly from A to Z with headless + query pipeline + trigger" and what not, I would keep for an "advanced usage" section of the documentation.

And, as Sami commented, to remove the `redirectTo` check for both example.

Reply:
I agree 🙂👍